src/Microsoft.AspNet.Server.Kestrel/Infrastructure/MemoryPoolIterator2.cs

```diff
@@ -180,58 +182,77 @@ public int Seek(Vector<byte> byte0Vector)
 
             var block = _block;
             var index = _index;
-            var array = block.Array;
+            var following = block.End - index;
+            byte[] array;
+            var byte0 = byte0Vector[0];
+
             while (true)
             {
-                while (block.End == index)
+                while (following == 0)
                 {
-                    if (block.Next == null)
+                    var newBlock = block.Next;
+                    if (newBlock == null)
                     {
                         _block = block;
                         _index = index;
                         return -1;
                     }
-                    block = block.Next;
-                    index = block.Start;
-                    array = block.Array;
+                    index = newBlock.Start;
+                    following = newBlock.End - index;
+                    block = newBlock;
                 }
-                while (block.End != index)
+                array = block.Array;
+                while (following > 0)
                 {
-                    var following = block.End - index;
-                    if (following >= Vector<byte>.Count)
+#if !DEBUG 
+// Need unit tests to test Vector path
+                    // Check will be Jitted away https://github.com/dotnet/coreclr/issues/1079
+                    if (Vector.IsHardwareAccelerated)
                     {
-                        var data = new Vector<byte>(array, index);
-                        var byte0Equals = Vector.Equals(data, byte0Vector);
-
-                        if (byte0Equals.Equals(Vector<byte>.Zero))
+#endif
+                        if (following >= _vectorSpan)
```

Comment:
`MemoryPoolIterator2Extensions.GetKnownString` also has this problem with `BitConverter.IsLittleEndian`

```
   285:             // This optimization only works on little endian environments (for now).
   286:             if (!BitConverter.IsLittleEndian)
00007FFEA7EFEBB9  mov         rcx,7FFEA7B85380h  
00007FFEA7EFEBC3  mov         edx,3  
00007FFEA7EFEBC8  call        00007FFF079EBC50  
00007FFEA7EFEBCD  mov         rax,7FFEA7B853D2h  
00007FFEA7EFEBD7  movzx       eax,byte ptr [rax]  
00007FFEA7EFEBDA  test        eax,eax  
00007FFEA7EFEBDC  jne         00007FFEA7EFEBE8  
```